test/helpers/kubectl.go

```diff
@@ -618,6 +671,98 @@ func (kub *Kubectl) CiliumPolicyAction(namespace, filepath string, action Resour
 	return "", nil
 }
 
+// CiliumExportInfo runs a subroutine in the background, that exports Cilium's
+// memory usage and the given commands to the Prometheus server. The function
+// also receives a context to stop the function when needed and a prefix to
+// store to send the data to the database. This function gathers the info every
+// 5 seconds.
+func (kub *Kubectl) CiliumExportInfo(ctx context.Context, prefix string, cmds map[string]string) {
+	tickerSleep := 5 * time.Second
+
+	getKey := func(metric string) string {
+		return fmt.Sprintf("%s_%s", prefix, metric)
+	}
+
+	ExecReport := func() config.PrometheusMetrics {
+		result := config.PrometheusMetrics{}
+		memory, err := kub.CiliumReportMemory()
+		if err != nil {
+			kub.logger.WithError(err).Warn("cannot get memory report")
+		}
+
+		// Memory report first
+		for k, v := range memory {
+			result[getKey(k)] = fmt.Sprintf("%d", v)
+		}
+
+		for k, v := range cmds {
+			res := kub.Exec(v)
+			if !res.WasSuccessful() {
+				kub.logger.WithError(fmt.Errorf("%s", res.CombineOutput())).Warnf(
+					"cannot export metric '%s' with cmd '%s'", k, v)
+			}
+			result[getKey(k)] = res.SingleOut()
+		}
+		return result
+	}
+
+	ticker := time.NewTicker(tickerSleep)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				data := ExecReport()
+				err := config.PushInfo(&data)
+				if err != nil {
+					kub.logger.WithError(err).Warn("cannot push info to Prometheus")
+				}
+			case <-ctx.Done():
+				return
```

Comment:
add `ticker.Stop()`

Reply:
Not sure why a ticker.Stop should be in there, it's already finished by context no?

Reply:
I think the ticker will keep running unless it's stopped